### ISSUE TITLE
Bottom navigation bar with navigationService - view doesn't maintain state

### ISSUE THREAD

**Issue by dvagala** [Stacked-Org/stacked#720]
In the [new bottom navigation bar example](https://github.com/FilledStacks/stacked/blob/master/packages/stacked/example/lib/ui/bottom_nav/bottom_nav_example.dart) where the `navigationService` is used to switch the pages, the view does no longer maintain it's state with [PageStorageKey](https://www.filledstacks.com/post/bottom-navigation-with-stacked-architecture/#the-view-does-not-maintain-state).

I like this new way of switching the pages with navigationService. Is there any way to preserve the view state?

**Comment by Eimen2018** [comment]
@dvagala try adding `with AutomaticKeepAliveClientMixin` on the state
` class BottomNavExampleState extends State<BottomNavExample> with AutomaticKeepAliveClientMixin`

and add 
```
@override
 bool get wantKeepAlive => true;
```

**Comment by dvagala** [comment]
@Eimen2018 I'm trying and no success

**Comment by Eimen2018** [comment]
I guess the other option is to not dispose the viewmodel of the data you want to page store at.

**Comment by dvagala** [comment]
Yes, but it won't preserve the pure `view state`. I mean things like a scroll position in ScrollView. I tried also storing the `scrollController` in the viewModel, but apparently it won't restore the scroll after the scroll view is reattached from the controller.

I think the main issue why the `PageStorageKey` doesn't work is that when switching the routes with `ExtendedNavigator`, there isn't a common `PageStorage` ancestor. I tried to play around with the router code generation but with no luck
